Add missing writer.finish() in annotate and alias commands

kimrutherford opened the pull request:
This fixes invalid JSON output from the annotate and alias commands.  The closing "]" was missing with "-O json".

Both of these commands produce invalid JSON:

```sh
runoak -i bioportal: annotate -O json "enlarged nucleus in T-cells from peripheral blood"
```

```sh
runoak -i sqlite:obo:fypo aliases -O json FYPO:0009021
```

This PR is my guess at a fix.

kimrutherford commented:
I've added more commits that add `writer.finish()` to `mappings()`, `ancestors()`, `descendants()` and `similarity_pair()` in `cli.py`.

Thanks for the great project!  I think it will be very useful for PomBase.

cmungall commented:
@hrshdhgd do you know what is going on here?

```
=================================== FAILURES ===================================
_______________ TestCommandLineInterface.test_mappings_curie_map _______________

self = <tests.test_cli.TestCommandLineInterface testMethod=test_mappings_curie_map>

    def test_mappings_curie_map(self):
        mappings_output = OUTPUT_DIR.joinpath("test_mappings.tsv")
        if os.name == "nt":
            shell = True
        else:
            shell = False
        result = subprocess.run(
            [
                "runoak",
                "-i",
                f"sqlite:{TEST_DB}",
                "mappings",
                "-O",
                "sssom",
                "-o",
                mappings_output,
            ],
            shell=shell,  # noqa
        )
        self.assertEqual(0, result.returncode)
>       msdf = parse_sssom_table(mappings_output)

tests/test_cli.py:418: 
```

hrshdhgd commented on src/oaklib/cli.py:
```diff
@@ -3466,6 +3470,7 @@ def mappings(terms, maps_to_source, autolabel: bool, output, output_type):
                 if autolabel:
                     impl.inject_mapping_labels([mapping])
                 writer.emit(mapping)
+    writer.finish()
```
@cmungall : Addition of this line creates the error you see in gh actions

hrshdhgd replied:
The output file is blank and hence the file parsed back contains nothing.

cmungall replied:
can you go ahead and make the change?

cmungall replied:
oh I see this is on a fork... we can get @kimrutherford set up with permissions to make branches off origin if that fits out flow better

kimrutherford commented:
> we can get @kimrutherford set up with permissions to make branches off origin if that fits out flow better

I'm happy to help.  If that's easier for you, please add me.

I've just added a minimal test for `runoak mappings` using JSON as the output format.

hrshdhgd commented:
I have added you @kimrutherford , kindly confirm!

kimrutherford commented:
> I have added you @kimrutherford , kindly confirm!

Thanks.  I've rebased my changes and made a branch in the main repo:
https://github.com/INCATools/ontology-access-kit/tree/kmr-json-write-fix

If you'd like me to close this PR and make a new one with the new branch, let me know.

hrshdhgd commented:
I think making a new PR with a branch within this repo would be helpful. That way I could quickly fix the obvious errors in the GitHub Actions QC and merge the branch asap.

hrshdhgd commented:
I think making a new PR with a branch within this repo would be helpful. That way I could quickly fix the obvious errors in the GitHub Actions QC and merge the branch asap.

kimrutherford commented:
I've open #501 to replace this PR.
